src/prefect/server/models/deployments.py

```diff
@@ -734,7 +734,7 @@ async def _generate_scheduled_flow_runs(
                     "work_queue_id": deployment.work_queue_id,
                     "parameters": parameters,
                     "infrastructure_document_id": deployment.infrastructure_document_id,
-                    "idempotency_key": f"scheduled {deployment.id} {date}",
+                    "idempotency_key": f"scheduled {deployment.id} {deployment_schedule.slug or deployment_schedule.id} {date}",
```

Comment:
For simplicity, I think we should use the schedule ID since we know that will always be populated.
```suggestion
                    "idempotency_key": f"scheduled {deployment.id} {deployment_schedule.id} {date}",
```

Reply:
good call - updated in [a90ebf5](https://github.com/PrefectHQ/prefect/pull/17123/commits/a90ebf51582c8a274dab5f9d8be539f1d49273f0)!